src/options.c

```diff
@@ -348,6 +348,11 @@ static void mi_error_default(int err) {
     abort();
   }
 #endif
+#if MI_XMALLOC
```

Comment:
Let's change this to `#if defined(MI_XMALLOC)` then

Reply:
Just wanted to ask: is this a project-wide decision to pass all defines via a cmdline? I was thinking also about the opposite approach, to have some cmake-generated config.h that would provide these defines. Both approaches have pros and cons

Reply:
Well, the original design goal was to have no configuration at all, just debug and release. Later there was secure, and now there are already a bunch of build options :-(  I would like to keep it as small as possible -- if we need cmake generated configs that would main I failed at that :-)  I think we keep it as it is for now and if possible, even remove/merge further options.